subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -125,16 +131,34 @@ static bool validation_info_check(const struct fw_validation_info *vinfo)
 
 /* Find the validation_info at the end of the firmware. */
 static const struct fw_validation_info *
-validation_info_find(u32_t start_address, u32_t search_distance)
+validation_info_find(u32_t start_address, u32_t search_distance,
+		     const struct fw_info *finfo)
 {
 	const struct fw_validation_info *vinfo;
 
+	/* The validation info is stored after the firmware image */
+	start_address += finfo->size;
+
 	for (int i = 0; i <= search_distance; i++) {
 		vinfo = (const struct fw_validation_info *)(start_address + i);
 		if (validation_info_check(vinfo)) {
 			return vinfo;
 		}
 	}
+
+	if (finfo->next_img_offset != 0) {
+		const struct fw_info *next_info =
+			fw_info_find(finfo->address + finfo->next_img_offset);
```

Comment:
All the intermediate fw_infos should be put through the checks in bl_validate_firmware(). Most importantly, check that all firmwares and firmware infos are within the signed region.